### PR TITLE
Move to TypeScript

### PR DESCRIPTION
**Pull request by Gnuk**

### CODE REVIEW

File: tech/express.ts [hunk 1]
```diff
@@ -1,16 +1,19 @@
-const express = require('express');
-const cors = require('cors')
-const app = express();
+import * as express from 'express';
 const reload = require('reload');
 const watch = require('node-watch');
```
Comment (nonomoho) [inline tech/express.ts]: Why not use `import * as ... from '...';`?

File: tech/filters/component-render.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { docUtils } from './doc-utils';
+import { Options, Render } from './Documentation';
+
+const render = (height: string): Render => (src) => `<iframe class="doc-iframe" src="${src}" height="${height}"></iframe>`;
+
+export const componentRender = (str: string, opts: Options): string => {
```
Comment (jeremietisserand) [inline tech/filters/component-render.ts]: What is the purpose of render only ? 

Do we need to keep this filter ?

File: tech/express.ts (hunk 1 above)
Comment (Gnuk) [reply tech/express.ts]: I'm waiting https://github.com/yuanchuan/node-watch/pull/87 and some dependencies have no types, I'll check if I can contribute to add types or if I should use another dependency for the others.

File: tech/filters/component-render.ts (hunk 2 above)
Comment (Gnuk) [reply tech/filters/component-render.ts]: I don't understand, can you be more explicit, maybe by using an example ?

File: tech/express.ts (hunk 1 above)
Comment (Gnuk) [reply tech/express.ts]: Ok for this import, I know there is other require in code to move.

**Comment by Gnuk** [comment]
Hi @tikui/developers 

I suggest to not fix `recursive-copy`, `reload` and `html2pug` currently because there is no types from these libraries.

I suggest to try to use `https://www.npmjs.com/package/fs-extra` instead of `recursive copy if there is an interesting filter in another PR.

If you want to add types to html2pug and reload, feel free to do it :D .

File: tech/filters/component-render.ts (hunk 2 above)
Comment (nonomoho) [reply tech/filters/component-render.ts]: View with @Gnuk, we close this conversation

**Pull request merged**